src/app/beer_garden/files.py

```diff
@@ -0,0 +1,606 @@
+    """
+    Deletes a file and its chunks.
+
+    Parameters:
+        file_id: The id of the file.
+    """
+    return _delete_file(file_id)
+
+
+def set_owner(file_id: str, owner_id: str = None, owner_type: str = None):
+    """
+    Sets the owner field of the file.  This is used for DB pruning.
+
+    Parameters:
+        file_id: The id of the file.
+        owner_id: The id of the owner.
+        owner_type: The type of the owner (job/request).
+    """
+    if (owner_id is not None) and (owner_type is not None):
+        file = check_file(file_id)
+        old_owner_priority = OWNERSHIP_PRIORITY.get(file.owner_type, 1_000_000)
+        new_owner_priority = OWNERSHIP_PRIORITY.get(owner_type, 1_000_000)
+        # Job owners should override request owners (if one is already set)
+        if file.owner_type is None or new_owner_priority <= old_owner_priority:
+            if owner_type in OWNERSHIP_MAP:
+                owner = db.query_unique(OWNERSHIP_MAP[owner_type], id=owner_id)
+                db.modify(
+                    file,
+                    owner_id=owner_id,
+                    owner_type=owner_type,
+                    owner=owner.id if owner is not None else None,
+                )
+            else:
+                db.modify(file, owner_id=owner_id, owner_type=owner_type)
+            # Refresh data
+            file = check_file(file.id)
+            return FileStatus(
+                operation_complete=True,
+                **_unroll_object(file, key_map={"id": _format_id}, ignore=["owner"]),
+            )
+        return FileStatus(
+            operation_complete=False,
+            message=f"Owner type {owner_type} has lower priority than {file.owner_type}",
+        )
+    return FileStatus(
+        operation_complete=False,
+        message=f"Operation FILE_OWN requires an owner type "
+        f"and id. Got {owner_type} and {owner_id}",
+    )
+
+
+def set_owner_for_files(owner_id, owner_type, parameters):
+    """ Scans the parameters for file ids, and sets ownership metadata."""
+    for id in _check_file_ids(parameters):
+        router.route(
+            Operation(
+                operation_type="FILE_OWNER",
+                args=[id],
+                kwargs={"owner_id": owner_id, "owner_type": owner_type},
+            )
+        )
+
+
+def _check_file_ids(parameter, ids=None):
+    """Used to scan operations for the FileID prefix.
+    Parameters:
+        parameter: The object to be scanned.
+        ids: The current list of discovered file ids
+    Returns:
+        A list of file ids (may be empty).
+    """
+    if ids is None:
+        ids = []
+    if isinstance(parameter, six.string_types):
+        if UI_FILE_ID_PREFIX in parameter:
+            try:
+                tmp_list = parameter.split(" ")
+                prefix_idx = parameter.index(UI_FILE_ID_PREFIX)
+                ids.append(tmp_list[prefix_idx + 1])
+            except (IndexError, ValueError):
+                pass
```

Comment:
What happens if some jerk decided to set their value to something that looks like the `UI_FILE_ID_PREFIX`? Can't you just check they key value against the System/Command object to see if it is a file type parameter?

Reply:
Looks like I don't have access to the information via the event payload (checked on debugger).  If someone really wants to replicate the prefix, we do checking to validate the ID and raise errors if the file isn't found.

Reply:
@TheBurchLog Is there a way to get this from the `Brewtils` models? The only place I see this information available is in the `Parameter` model and I'm not sure how I'm supposed to access that given the `Request` or `Job` model.

Reply:
@SpecificallyGravity Ok, so I did some more thinking on this. The fact that we allow complex models that nest parameters means that we need this level of recursion. If we didn't want to iterate through all of the parameters, the only way we could do this is to grab the Command Object. This would be done like so:

```python
request_system = systems.get_systems(
            namespace=operation.model.namespace,
            name=operation.model.system,
            version=operation.model.system_version,
        )
if len(request_systems) == 1:
    request_system = request_systems[0]
    For loop over commands for command_name = operation.command_name
        for loop over parameters 
            if type is base64, process file
            if parameter has parameters, loop through complex object parameters, if match, process file

```

Now I am not sure if this is really worth it because it would probably be faster to do it the way you are doing it. That way we don't have to do crazy mapping at the end for weird complex objects.